src/compiler/core.ts

```diff
@@ -1985,26 +1970,30 @@ namespace ts {
         let previous = new Array(s2.length + 1);
         let current = new Array(s2.length + 1);
         /** Represents any value > max. We don't care about the particular value. */
-        const big = max + 1;
+        const big = max + 0.000000001;
 
         for (let i = 0; i <= s2.length; i++) {
             previous[i] = i;
         }
 
         for (let i = 1; i <= s1.length; i++) {
             const c1 = s1.charCodeAt(i - 1);
-            const minJ = i > max ? i - max : 1;
-            const maxJ = s2.length > max + i ? max + i : s2.length;
+            const minJ = Math.ceil(Math.max(i - max, 1));
+            const maxJ = Math.floor(Math.min(max + i, s2.length));
             current[0] = i;
             /** Smallest value of the matrix in the ith column. */
             let colMin = i;
             for (let j = 1; j < minJ; j++) {
                 current[j] = big;
             }
             for (let j = minJ; j <= maxJ; j++) {
+                // case difference should be insignificant.
+                const substitutionDistance = s1[i - 1].toLowerCase() === s2[j-1].toLowerCase()

```

Comment:
Would it make sense to cache the lowercase versions of `s1` and `s2` and index into those instead?

This piece of code needs a microbenchmark, since I have no idea. I'm not sure how to do that off the top of my head though.

Reply:
How about extracting all symbols from some mostly used type definitions and then apply `getSpellingSuggestion` randomly?

Reply:
Maybe? One way to stress the system would be to have a scope that fills in all possible neighbours, so that the algorithm has to run on lots of candidates, all under the maximum distance cutoff.

I'm going to run the perf suite on this too, since all those projects compile with lots of errors now -- maybe enough of them are name resolution errors to notice a slowdown.